### PR TITLE
Set a default key and value for jsbox config

### PR DESCRIPTION
**Pull request by justinvdm**
**key:**
config

**value:**
{
    "name": "<slugified-conversation-name>"
}

---

Friends with #965?

### CODE REVIEW

**Comment by justinvdm** [comment]
@hodgestar ready for review.

There seems to be quite a bit of duplication in the code for creating wizards and the create conversation view, this PR just adds to that :/ not sure if cleaning that up is something for this PR (if it is even worth it at this point).

The tests were a similar case. The wizard create view's tests use actual conversation types while the create conversation view tests use dummy conversation types. I ended up getting the wizard create view's tests to know about dummy conversation types too in order to get this feature tested, but it wasn't pretty :/ Let me know if you think there are ways of cleaning this up that are worth it.

File: go/wizard/views.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ def post(self, request):
         if not all(forms_valid):
             logger.info("Validation failed: %s" % (
                 [frm.errors for frm in forms_to_validate],))
+            print frm.errors
```
Comment (hodgestar) [inline go/wizard/views.py]: Stray print.

**Comment by hodgestar** [comment]
I notice we're not currently slugifying the conversation name to generate the default config (we're just using the static string `conv-name`. Is the slugifying something we want to do? Also happy for that to happen in a separate PR if we do want to do it.

**Comment by justinvdm** [comment]
@hodgestar yup, noticed that too, should've probably mentioned something. That problem hurts us for user-defined names too, so I think that change should live in the toolkit.

**Comment by justinvdm** [comment]
@hodgestar Ah, that should not be a static string, sorry :/ fixing

**Comment by justinvdm** [comment]
@hodgestar point about the config 'name' field being parsed inside the toolkit still applies though (the static string was a different thing).

**Comment by hodgestar** [comment]
I think I'd be happier to use `slugify` (`from django.template.defaultfilters import slugify`) here on the conversation name because this is meant to be developer facing config and I (with my sandbox user hat on) think I'd rather the sandbox didn't mangle the name unnecessarily. Ideally validation would provide early feedback if the config is invalid (although we don't have that yet).

**Comment by justinvdm** [comment]
@hodgestar Cool, validation for the config sounds better. Would django's slugify work though? Can't remember if our metric names can contain `-`s.

**Comment by hodgestar** [comment]
The regex is `^[a-zA-Z][a-zA-Z0-9._]{,100}$` but maybe we should just add
dashes to the allowed list of characters?

**Comment by justinvdm** [comment]
+1, though that means keeping this open until we have graphite understanding that. Okay with that though. Will add the slugifying so long.

**Comment by SteveBarnett** [comment]
:+1: Like @hodgestar [said](https://github.com/praekelt/vumi-go/pull/1032#issuecomment-60754275): I think our developer persona (WIP!) would prefer slugified with dashes. :)

**Comment by justinvdm** [comment]
Ready for re-review, though I don't think we should land this until after the retreat. There's a chance graphite won't like our idea to use dashes.

**Comment by hodgestar** [comment]
:+1: and agreed on not landing until we're back.

**Comment by SteveBarnett** [comment]
\o/. Nice one, @justinvdm and @hodgestar.

**Pull request merged**